x-pack/plugins/security_solution/public/resolver/view/panel.tsx

```diff
@@ -3,191 +3,245 @@
+  const graphableProcessEntityIds = useMemo(() => {
+    return new Set(graphableProcesses.map(event.entityId));
+  }, [graphableProcesses]);
+  // The entity id in query params of a graphable process (or false if none is found)
+  // For 1 case (the related detail, see below), the process id will be in crumbEvent instead of crumbId
+  const idFromParams = useMemo(() => {
+    if (graphableProcessEntityIds.has(queryParams.crumbId)) {
+      return queryParams.crumbId;
+    }
+    if (graphableProcessEntityIds.has(queryParams.crumbEvent)) {
+      return queryParams.crumbEvent;
+    }
+    return '';
+  }, [queryParams, graphableProcessEntityIds]);
 
-  const formatter = new Intl.DateTimeFormat(i18n.getLocale(), {
-    year: 'numeric',
-    month: '2-digit',
-    day: '2-digit',
-    hour: '2-digit',
-    minute: '2-digit',
-    second: '2-digit',
-  });
+  // The "selected" node (and its corresponding event) in the tree control.
+  // It may need to be synchronized with the ID indicated as selected via the `idFromParams`
+  // memo above. When this is the case, it is handled by the layout effect below.
+  const selectedDescendantProcessId = useSelector(selectors.uiSelectedDescendantProcessId);
+  const uiSelectedEvent = useMemo(() => {
+    return graphableProcesses.find((evt) => event.entityId(evt) === selectedDescendantProcessId);
+  }, [graphableProcesses, selectedDescendantProcessId]);
 
-  const dispatch = useResolverDispatch();
+  // Until an event is dispatched during update, the event indicated as selected by params may
+  // be different than the one in state.
+  const paramsSelectedEvent = useMemo(() => {
+    return graphableProcesses.find((evt) => event.entityId(evt) === idFromParams);
+  }, [graphableProcesses, idFromParams]);
+  const { timestamp } = useContext(SideEffectContext);
+  const [lastUpdatedProcess, setLastUpdatedProcess] = useState<null | ResolverEvent>(null);
 
-  const handleBringIntoViewClick = useCallback(
-    (processTableViewItem) => {
+  /**
+   * When the ui-selected node is _not_ the one indicated by the query params, but the id from params _is_ in the current tree,
+   * dispatch a selection action to amend the UI state to hold the query id as "selected".
+   * This is to cover cases where users e.g. share links to reconstitute a Resolver state or
+   * an effect pushes a new process id to the query params.
+   */
+  useLayoutEffect(() => {
+    if (
+      paramsSelectedEvent &&
+      // Check state to ensure we don't dispatch this in a way that causes unnecessary re-renders, or disrupts animation:
+      paramsSelectedEvent !== lastUpdatedProcess &&
+      paramsSelectedEvent !== uiSelectedEvent
+    ) {
+      setLastUpdatedProcess(paramsSelectedEvent);
       dispatch({
         type: 'userBroughtProcessIntoView',
         payload: {
           time: timestamp(),
-          process: processTableViewItem.event,
+          process: paramsSelectedEvent,
```

Comment:
re: our discussion. Could you change this to a new action that starts w/ 'app'? ty

Reply:
https://github.com/elastic/kibana/pull/69098/commits/a7cdb88ab14e42d111cca2e3d8ab24270c673248